### PR TITLE
[UIKit] Partial update to Xcode 11 Beta 1 and 2 - Part 1 of ?

### PR DESCRIPTION
**Pull request by dalexsoto**
Adds new classes included in Xcode 11 Beta 1 and 2, all stuff
included in this PR is up to date with Beta 2.

### CODE REVIEW

File: src/uikit.cs [hunk 1]
```diff
@@ -863,6 +863,11 @@ partial interface NSTextAttachment : NSTextAttachmentContainer, NSCoding
 		[NullAllowed] // by default this property is null
 		[Export ("fileWrapper", ArgumentSemantic.Retain)]
 		NSFileWrapper FileWrapper { get; set; }
+
+		[Watch (6,0), TV (13,0), iOS (13,0)]
+		[Static]
+		[Export ("textAttachmentWithImage:")]
+		NSTextAttachment FromImage (UIImage image);
```
Comment (rolfbjarne) [inline src/uikit.cs]: No verb in method name.

File: src/uikit.cs [hunk 2]
```diff
@@ -6712,6 +6812,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+	}
+
+	[iOS (13,0), NoWatch]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UISceneConfiguration : NSCopying, NSSecureCoding {
+
+		[Static]
+		[Export ("configurationWithName:sessionRole:")]
+		UISceneConfiguration FromName ([NullAllowed] string name, [BindAs (typeof (UISceneSessionRole))] NSString sessionRole);
```
Comment (rolfbjarne) [inline src/uikit.cs]: No verb in method name.

**Comment by dalexsoto** [comment]
@rolfbjarne I agree on the naming of the non-static method naming I will revisit them, about the static ones we have the `FromFoo` Naming pattern everywhere see for example https://docs.microsoft.com/en-us/dotnet/api/uikit.uiimage.frombundle?view=xamarin-ios-sdk-12 so I think keeping the static factory methods being named `From` is fine and consistent with what we currently do

What do you think?

File: src/uikit.cs (hunk 1 above)
Comment (dalexsoto) [reply src/uikit.cs]: We have other instances of from usage

* NSLayoutConstraint [] FromVisualFormat
* UIApplicationShortcutIcon FromType
* UIApplicationShortcutIcon FromTemplateImageName
* UIApplicationShortcutIcon FromContact
* UIColor FromWhiteAlpha
* UIColor FromCGColor

and many others, I think this one is fine, any `From` usage is fine in my opinion as long as it is a Static method and returns the container type.

File: src/uikit.cs (hunk 2 above)
Comment (dalexsoto) [reply src/uikit.cs]: Same justification https://github.com/xamarin/xamarin-macios/pull/6392/files#r296216711

File: src/uikit.cs [hunk 3]
```diff
@@ -7127,6 +7483,287 @@ interface UIImage : NSSecureCoding
...
+		[Export ("systemImageNamed:compatibleWithTraitCollection:")]
+		[return: NullAllowed]
+		UIImage GetSystemImage (string name, [NullAllowed] UITraitCollection traitCollection);
+#endif
+		[Watch (6,0), TV (13,0), iOS (13,0)]
+		[Static]
+		[ThreadSafe]
+		[Export ("imageNamed:inBundle:withConfiguration:")]
+		[return: NullAllowed]
+		UIImage FromBundle (string name, [NullAllowed] NSBundle bundle, [NullAllowed] UIImageConfiguration configuration);
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/uikit.cs]: Same justification as https://github.com/xamarin/xamarin-macios/pull/6392/files#r296216711 also this ends up being an overload of an existing method

File: src/uikit.cs [hunk 4]
```diff
@@ -7127,6 +7483,287 @@ interface UIImage : NSSecureCoding
...
+	[BaseType (typeof (UIImageConfiguration))]
+	interface UIImageSymbolConfiguration {
+
+		[Static]
+		[Export ("unspecifiedConfiguration")]
+		UIImageSymbolConfiguration UnspecifiedConfiguration { get; }
+
+		[Static]
+		[Export ("configurationWithScale:")]
+		UIImageSymbolConfiguration FromScale (UIImageSymbolScale scale);
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/uikit.cs]: Same justification as https://github.com/xamarin/xamarin-macios/pull/6392/files#r296216711 but I also like your Create suggestion...

I think it is the same case like `NSNumber` you can find what you are looking at a single glance and kinda matches a little bit more if you are coming from the Objective-C documentation  

image

image

I'm thinking on keeping the `From` methods, I think it is much more discoverable this way and easier if you are coming from the docs.

File: src/uikit.cs [hunk 5]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Export ("openURL:options:completionHandler:")]
+		void OpenUrl (NSUrl url, [NullAllowed] UISceneOpenExternalUrlOptions options, [NullAllowed] Action<bool> completion);
+
+		[Export ("title")]
+		string Title { get; set; }
+
+		[Export ("activationConditions", ArgumentSemantic.Strong)]
+		UISceneActivationConditions ActivationConditions { get; set; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: unneeded, it's already on the type itself

File: src/uikit.cs [hunk 6]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Field ("UISceneWillConnectNotification")]
+		[Notification]
+		NSString WillConnectNotification { get; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 7]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Field ("UISceneDidDisconnectNotification")]
+		[Notification]
+		NSString DidDisconnectNotification { get; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 8]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Field ("UISceneDidActivateNotification")]
+		[Notification]
+		NSString DidActivateNotification { get; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 9]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Field ("UISceneWillDeactivateNotification")]
+		[Notification]
+		NSString WillDeactivateNotification { get; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 10]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+		[Field ("UISceneWillEnterForegroundNotification")]
+		[Notification]
+		NSString WillEnterForegroundNotification { get; }
+
+		[iOS (13,0)]
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 11]
```diff
@@ -6712,6 +6813,262 @@ interface UIScreenEdgePanGestureRecognizer {
...
+	}
+
+	[iOS (13,0), TV (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UISceneActivationConditions : NSSecureCoding {
+
+		[Export ("init")]
+		[DesignatedInitializer]
+		IntPtr Constructor ();
```
Comment (spouliot) [inline src/uikit.cs]: there's a class-level `[DesignatedDefaultCtor]`

File: src/uikit.cs [hunk 12]
```diff
@@ -7127,6 +7484,287 @@ interface UIImage : NSSecureCoding
...
+	}
+
+	[NoWatch, NoTV, iOS (13,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface UIPreviewParameters : NSCopying {
+
+		[Export ("init")]
+		[DesignatedInitializer]
+		IntPtr Constructor ();
```
Comment (spouliot) [inline src/uikit.cs]: same

File: src/uikit.cs [hunk 13]
```diff
@@ -13124,6 +13775,11 @@ interface UIView : UIAppearance, UIAppearanceContainer, UIAccessibility, UIDynam
 		[Async]
 		void PerformSystemAnimation (UISystemAnimation animation, UIView [] views, UIViewAnimationOptions options, Action parallelAnimations, UICompletionHandler completion);
 
+		[TV (12,0), iOS (12,0)] // Yep headers state iOS 12
```
Comment (spouliot) [inline src/uikit.cs]: make sure to run intro tests on an iOS 12.0 simulator :)

File: src/UIKit/UIEnums.cs [hunk 14]
```diff
@@ -2272,4 +2279,197 @@ public enum UIPrintErrorCode : long
...
+	[NoWatch, NoTV, iOS (13, 0)]
+	[Native]
+	public enum UIContextMenuInteractionCommitStyle : long
+	{
+		Dismiss = 0,
+		Pop,
+	}
+
+	[iOS (13,0), TV (13,0), NoWatch]
+	public enum UISceneSessionRole
```
[the remark this answers is not in the document]
Comment (dalexsoto) [reply src/UIKit/UIEnums.cs]: ¯\_(ツ)_/¯ nope I actually like it more, unsure how I got there :P

**Comment by rolfbjarne** [comment]
@dalexsoto one answer to your question is that past mistakes don't excuse us from doing the right thing in the future :smile:

Personally I like the `Create` methods (which we've used a lot too), but the `From*` isn't that bad.

@spouliot what do you think?

**Comment by spouliot** [comment]
I prefer the `Create` (any name with a verb for methods) but we have a lot of `From` (not the worse non-verb case) and, at least for existing types, we should be consistent.

**Comment by dalexsoto** [comment]
build

**Comment by dalexsoto** [comment]
Fixed Apple TV availability, Watch was fine with intro tests on 5.0.

**Pull request merged**